src/hooks/useConnectRelatedVenues.ts

```diff
@@ -235,11 +259,34 @@ export const useConnectRelatedVenues: ReactHook<
     [currentVenue, parentVenue, siblingVenues, subvenues]
   );
 
+  const isRelatedVenuesLoaded = useMemo(() => {
+    const loaders = [
+      isCurrentVenueLoaded,
+      isSubvenuesLoaded,
+      isSiblingVenuesLoaded,
+    ];
+
+    if (parentId) {
+      loaders.push(isParentVenueLoaded);
+    }
+
+    return loaders.reduce(
+      (finalIsLoaded, isLoaded) => finalIsLoaded && isLoaded
+    );
+  }, [
+    parentId,
+    isCurrentVenueLoaded,
+    isParentVenueLoaded,
+    isSubvenuesLoaded,
+    isSiblingVenuesLoaded,
+  ]);
```

Comment:
I had exactly the same approach on this, but then walked through edge cases and thought that it is inappropriate to have `isParentVenueLoaded` true => false => true. But eventually, the result would be the same.

Reply:
Since your approach looks neater and mine is easier in terms of brain efforts, I'd vote for choosing mine @0xdevalias

Reply:
My concerns were more from a performance aspect, we're likely needlessly adding the performance hit of using `useMemo`: the extra comparisons to check the 'deps' on each render + the related memory storage. I'm not sure what optimisations (if any) happen as part of our transpilation, but I would imagine your way would also remove the ability for those sorts of things to run.

In the grand scheme of things, I expect this isn't going to be a dramatic impact on performance/memory usage.. so if you feel really strongly about keeping it your way, i'm not against it. But I do feel like it's a needless 'premature optimisation' that may actually hurt our performance (even if it's not noticeable at a 'real world' level)

Reply:
Subjective opinion, but i'm also not sure that yours is necessarily easier on the 'brain efforts' as you suggest, as you need to think about the array mutations, and then reducing over the array; compared to just `&&`'ing a bunch of `boolean`s together.

Reply:
Will probably keep it as-is. Doesn't seem to be very important

Reply:
The 'importance' is that we're needlessly adding memoisation (which will use more CPU to check whether the memo is broken, as well as more memory to store the result) seemingly only to change how the code is laid out slightly.

I feel pretty strongly about this.

Reply:
Changed this since it is rather a small change. However, I don't think it is productive to spend time on such minor things since we have a lot of bugs to fix and features to implement. Honestly, I'm not sure memoisation can anyhow affect the performance, in this case, thus this seems rather unrelated `needlessly adding memoisation (which will use more CPU to check whether the memo is broken, as well as more memory to store the result)`.

The main reason I implemented it like that was to reduce the cognitive load(in my opinion it was easier to understand what was going, when we excplitily added isParentVenueLoaded, which is, I agree, highly subjective). I think this dispute is worth discussing on the retrospective

Reply:
Thanks for making the change. Happy to discuss this more during sprint retro.